editor/src/com/talosvfx/talos/editor/project2/apps/AppPrefs.java

```diff
@@ -0,0 +1,36 @@
+package com.talosvfx.talos.editor.project2.apps;
+
+import com.badlogic.gdx.utils.ObjectMap;
+import com.talosvfx.talos.editor.addons.scene.assets.GameAsset;
+
+/**
+ * App preferences per asset
+ */
+public class AppPrefs {
+    private ObjectMap<String, Object> preferences;
+
+    public AppPrefs () {
+        preferences = new ObjectMap<>();
+    }
+
+    public boolean hasPrefFor(GameAsset<?> gameAsset) {
+        return preferences.containsKey(getUUID(gameAsset));
+    }
+
+    public Object getPrefFor(GameAsset<?> gameAsset) {
+        return preferences.get(getUUID(gameAsset));
+    }
+
+    public void setPrefFor(GameAsset<?> gameAsset, Object appPref) {
+        preferences.put(getUUID(gameAsset), appPref);
+    }
+
+    private String getUUID (GameAsset<?> gameAsset) {
+        if (!gameAsset.dependentRawAssets.isEmpty()) {
+            return gameAsset.getRootRawAsset().metaData.uuid.toString();
```

Comment:
Also, with the fake game asset, we can probably make a fake dependent raw asset too, s we don't need this check

Reply:
Ive just done this for one of my tasks, so you can use the same

Reply:
Done